### PR TITLE
Structure convention on Core

### PR DESCRIPTION
**Pull request by michelguerin**
This PR contains the new structure convention and the logic needed by PR #74

### CODE REVIEW

File: doc/adr/0004-structure-convention.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+The core packages gains more and more logic, we need a way to have small piece of code that are easily maintainable.
+
+## Decision
+
+### Domain Driven
+Code is split by domain. Each domain will have its own directory, and within each directory, the code will be further organized by functionality. This structure will help in maintaining a clean and manageable codebase.
+
+### Naming Convention
+
+To maintain consistency across the codebase, we will adopt a clear and concise naming convention. All files and directories will use `kebab-case` for their names. Except for main or index file, each file will be named using the format `name-of-file.name-of-folder.ts`, ensuring uniformity and readability.
```
Comment (illia-solovei-s1) [inline doc/adr/0004-structure-convention.md:20]: Naming, could you provide more examples on this? For instance, how would you name a service file or a utility file within a domain?

Comment (illia-solovei-s1) [inline doc/adr/0004-structure-convention.md:20]: Could you provide more examples on this? For instance, how would you name a service file or a utility file within a domain?

File: doc/adr/0004-structure-convention.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+To maintain consistency across the codebase, we will adopt a clear and concise naming convention. All files and directories will use `kebab-case` for their names. Except for main or index file, each file will be named using the format `name-of-file.name-of-folder.ts`, ensuring uniformity and readability. 
+
+#### Examples
+
+- Directories: `data-type-templates`
+- Files: `queries.data-type-templates.ts`
+
+### Type Declaration Files
+
+Type declaration files are special TypeScript files that end with the `.d.ts` extension. These files are used exclusively for declaring types and do not contain any executable code. By using `.d.ts` files, we ensure that type definitions are centralized, easily accessible, and separate from implementation code.
```
Comment (illia-solovei-s1) [inline doc/adr/0004-structure-convention.md:29]: Should the `.d.ts` files be placed within the same domain directories, or should there be a centralized directory for all type declarations?

File: doc/adr/0004-structure-convention.md (hunk 1 above)
Comment (michelguerin) [reply doc/adr/0004-structure-convention.md:20]: As these files will be on the same folder as the domain, I think we can name them service.name-of-folder.ts or utils.name-of-folder.ts.
We switch the logic one can found by NestJS to have a direct view of the context of the file in small directory tree.
I'll add these examples to the ADR.

File: doc/adr/0004-structure-convention.md (hunk 2 above)
Comment (michelguerin) [reply doc/adr/0004-structure-convention.md:29]: Yep in the same directory as the domain. I'll update the ADR to make it more clear.

File: packages/core/src/scd-queries/data-type-templates/queries.data-type-templates.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	constructor(root: Element) {
+		super(root)
+		if (!ELEMENT_NAMES) {
+			throw new Error("ELEMENT_NAMES is not initialized");
+		}
+	}
+
+	public static dataTypesTemplatesTree: DataTypeTemplates.ElementsTreeStructure = {
+		substation: {
+			elementName: ELEMENT_NAMES.substation,
```
Comment (trusz) [inline packages/core/src/scd-queries/data-type-templates/queries.data-type-templates.ts:18]: maybe we should rename to `ELEMENT_NAMES.substationType`

File: packages/core/src/scd-queries/data-type-templates/queries.data-type-templates.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+export class DataTypeTemplatesQueries extends SCDQueries {
+	constructor(root: Element) {
+		super(root)
+		if (!ELEMENT_NAMES) {
+			throw new Error("ELEMENT_NAMES is not initialized");
+		}
+	}
+
+	public static dataTypesTemplatesTree: DataTypeTemplates.ElementsTreeStructure = {
```
Comment (trusz) [inline packages/core/src/scd-queries/data-type-templates/queries.data-type-templates.ts:16]: Michel, check if this is not redundant because of the XLST

**Pull request merged**